Replicator/Pusher.cc

```diff
@@ -512,26 +512,76 @@ namespace litecore::repl {
         Worker::_connectionClosed();
     }
 
-    bool Pusher::isBusy() const {
-        return Worker::computeActivityLevel() == kC4Busy || (_started && (!_caughtUp || !_continuousCaughtUp))
-               || _changeListsInFlight > 0 || _revisionsInFlight > 0 || _blobsInFlight > 0 || !_revQueue.empty()
-               || !_pushingDocs.empty() || _revisionBytesAwaitingReply > 0;
+    bool Pusher::isBusy(std::string* reason) const {
+        std::string parentReason;
+        auto        workerLevel = Worker::computeActivityLevel(reason ? &parentReason : nullptr);
+        bool        ret         = workerLevel == kC4Busy || (_started && (!_caughtUp || !_continuousCaughtUp))
+                   || _changeListsInFlight > 0 || _revisionsInFlight > 0 || _blobsInFlight > 0 || !_revQueue.empty()
+                   || !_pushingDocs.empty() || _revisionBytesAwaitingReply > 0;
+        if ( ret && reason ) {
+            if ( workerLevel == kC4Busy ) *reason = std::move(parentReason);
+            else if ( _started && (!_caughtUp || !_continuousCaughtUp) )
+                *reason = "notCaughtUp";
+            else if ( _changeListsInFlight > 0 )
+                *reason = "changeListsInFlight";
+            else if ( _revisionsInFlight > 0 )
+                *reason = "revisionsInFlight";
+            else if ( _blobsInFlight > 0 )
+                *reason = "blobsInFlight";
+            else if ( !_revQueue.empty() )
+                *reason = "revQueue";
+            else if ( !_pushingDocs.empty() )
+                *reason = "pushingDocs";
+            else
+                *reason = "revisionBytesAwaitingReply";
```

Comment:
I'm wondering if it's better to have reason = "Busy as revisionsInFlight = x, changeListsInFlight = y, ..." so that we also know their values as well.